app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,113 @@
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if !is_logged_in?
+      redirect '/login'
+    else
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    end
+  end
+
+  get '/errors/login' do
+    erb :'/errors/login'
+  end
+
+  get '/errors/signup' do
+    erb :'/errors/signup'
+  end
+
+  get '/errors/tweets_show' do
+    erb :'/errors/tweets_show'
+  end
+
+  get '/tweets/new' do
+    if !is_logged_in?
+      redirect '/login'
+    else
+      erb :'/tweets/new'
+    end
+  end
+
+  get '/tweets/show' do
+    if !is_logged_in?
+      redirect '/login'
+    else
+      erb :'/tweets/show'
+    end
+  end
+
+  get '/tweets/:id' do
+    if !is_logged_in?
+      redirect '/login'
+    else
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if !is_logged_in?
+      redirect '/login'
+    else
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == current_user.id
+        erb :'/tweets/edit'
+      else
+        erb :'/errors/tweets/show'
+      end
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] == ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.